netcontrold/tests/lib/test_data_coll.py

```diff
@@ -0,0 +1,287 @@
+ifindex             : 6054325
+ingress_policing_burst: 0
+ingress_policing_rate: 0
+lacp_current        : []
+link_resets         : 0
+link_speed          : 10000000000
+link_state          : up
+lldp                : {}
+mac                 : []
+mac_in_use          : "24:6e:96:c4:2f:ea"
+mtu                 : 1500
+mtu_request         : []
+name                : "port2"
+ofport              : 11
+ofport_request      : 11
+options             : {dpdk-devargs="0000:19:00.1", n_rxq="1", n_rxq_desc="4096", n_txq="1", n_txq_desc="4096"}
+other_config        : {}
+statistics          : {flow_director_filter_add_errors=0, flow_director_filter_remove_errors=0, mac_local_errors=1, mac_remote_errors=0, "rx_128_to_255_packets"=0, "rx_1_to_64_packets"=0, "rx_256_to_511_packets"=0, "rx_512_to_1023_packets"=0, "rx_65_to_127_packets"=0, rx_broadcast_packets=0, rx_bytes=0, rx_crc_errors=0, rx_dropped=0, rx_errors=0, rx_fcoe_crc_errors=0, rx_fcoe_dropped=0, rx_fcoe_mbuf_allocation_errors=0, rx_fragment_errors=0, rx_illegal_byte_errors=0, rx_jabber_errors=0, rx_length_errors=0, rx_mac_short_packet_dropped=0, rx_management_dropped=0, rx_management_packets=0, rx_mbuf_allocation_errors=0, rx_missed_errors=0, rx_oversize_errors=0, rx_packets=0, "rx_priority0_dropped"=0, "rx_priority0_mbuf_allocation_errors"=0, "rx_priority1_dropped"=0, "rx_priority1_mbuf_allocation_errors"=0, "rx_priority2_dropped"=0, "rx_priority2_mbuf_allocation_errors"=0, "rx_priority3_dropped"=0, "rx_priority3_mbuf_allocation_errors"=0, "rx_priority4_dropped"=0, "rx_priority4_mbuf_allocation_errors"=0, "rx_priority5_dropped"=0, "rx_priority5_mbuf_allocation_errors"=0, "rx_priority6_dropped"=0, "rx_priority6_mbuf_allocation_errors"=0, "rx_priority7_dropped"=0, "rx_priority7_mbuf_allocation_errors"=0, rx_undersize_errors=0, "tx_128_to_255_packets"=0, "tx_1_to_64_packets"=0, "tx_256_to_511_packets"=0, "tx_512_to_1023_packets"=0, "tx_65_to_127_packets"=0, tx_broadcast_packets=0, tx_bytes=0, tx_dropped=0, tx_errors=0, tx_management_packets=0, tx_multicast_packets=0, tx_packets=0}
+status              : {driver_name=net_ixgbe, if_descr="DPDK 18.11.2 net_ixgbe", if_type="6", link_speed="10Gbps", max_hash_mac_addrs="4096", max_mac_addrs="127", max_rx_pktlen="1518", max_rx_queues="128", max_tx_queues="64", max_vfs="0", max_vmdq_pools="64", min_rx_bufsize="1024", numa_id="0", pci-device_id="0x10fb", pci-vendor_id="0x8086", port_no="1"}
+type                : dpdk"""
+
+
+class TestDataif_Collection(TestCase):
+    """
+    Test for getting pmd stats.
+    """
+
+    # create an empty pmd_map
+    pmd_map = dict()
+
+    def setUp(self):
+        # turn off limited info shown in assert failure for pmd object.
+        self.maxDiff = None
+
+        # a noop handler for debug info log.
+        class NlogNoop(object):
+
+            def debug(self, *args):
+                None
+
+        dataif.Context.nlog = NlogNoop()
+
+        # create one pmd object.
+        self.fx_pmd_1 = dataif.Dataif_Pmd(1)
```

Comment:
You set Testcase attribute (fx_pmd_1) only when you would like to share it across its functions.
As you pack pmd obj in self.pmd_map, you do not need to set it as instance variable, but just a local variable.

Reply:
sorry.. I didnt understand

Reply:
is it about self. ?

Reply:
yes. why should u have it with obj instance (self) which is not required for ur case.